### PR TITLE
add sparse feature importance support for lightgbm surrogate model

### PR DESCRIPTION
**Pull request by imatiach-msft**
Note: this is using sparse logic added to lightgbm after 3.1.0 release

for lightgbm surrogate model in mimic explainer, if input data is sparse call predict directly instead of using shap TreeExplainer, which doesn't have sparse logic implemented yet

note: existing tests already validate sparse logic works correctly

### CODE REVIEW

File: python/interpret_community/mimic/models/lightgbm_model.py [hunk 1]
```diff
@@ -47,7 +48,9 @@ def __init__(self, function):
         self._function = function
 
     def predict_wrapper(self, X, *args, **kwargs):
-        """Wraps a prediction function from lightgbm learner to densify the input dataset.
+        """Wraps a prediction function from lightgbm learner.
+
+        If version is 3.0.0, densifies the input dataset.
```
Comment (gregorybchris) [inline python/interpret_community/mimic/models/lightgbm_model.py:53]: Nit: might want to specify the direction >= like you do below

File: python/interpret_community/mimic/models/lightgbm_model.py [hunk 2]
```diff
@@ -59,6 +62,69 @@ def predict_wrapper(self, X, *args, **kwargs):
...
+        """
+        if issparse(X):
+            shap_values = self._lgbm.predict(X,
+                                             num_iteration=self._num_iters,
+                                             pred_contrib=True)
+            if isinstance(shap_values, list):
+                shape = shap_values[0].shape
+                self.expected_value = shap_values[0][0, shape[1]-1]
+                for idx, class_values in enumerate(shap_values):
+                    shap_values[idx] = class_values[:, :shape[1]-1]
```
Comment (gregorybchris) [inline python/interpret_community/mimic/models/lightgbm_model.py:117]: Nit: spaces around `-` operator (also see above and below), I'm surprised linting builds didn't catch this?

**Review by gregorybchris — APPROVED** [review]

File: python/interpret_community/mimic/models/lightgbm_model.py (hunk 1 above)
Comment (imatiach-msft) [reply python/interpret_community/mimic/models/lightgbm_model.py:53]: done, added ==

File: python/interpret_community/mimic/models/lightgbm_model.py (hunk 2 above)
Comment (imatiach-msft) [reply python/interpret_community/mimic/models/lightgbm_model.py:117]: interesting, running:

```
flake8 --max-line-length=119
```

seems to pass either way.  Anyway I've updated it to have spaces, thanks!

**Pull request merged**